### PR TITLE
Remove: buying/selling/owning company shares

### PR DESCRIPTION
**Pull request by TrueBrain**
## Motivation / Problem

Right, where to start.

First off, a personal opinion: this feature is just the worst. 

Let me explain.

Shares, as implemented, is just very poor, and makes no sense at all. You can buy shares of another company, and sell it. But when you sell it, nobody else has to buy it. So you either generate money from thin air, or push it in the aether. That is not how that works.
But also .. why 25%? Why not .. any percentage? Why not an amount?
From an economical point of view, it is just very .. weird.

Codewise, it is just a constant stream of problems. So many bug-reports over the years, either because we calculate the shares with the company value, or we don't, or it is mismatching, or or or or .. it is just a nightmare.

Mostly, you can generate infinite money with this. #7691 as example.

Now, what does it add to the player?

ABSOLUTELY NOTHING.

In fact, we have it disabled by default. Additionally, you could buy out an AI company in full, but not a human company ..
It is just so weird.

And the list goes on ... and on ...

It is just a bad feature.

So that gives us two choices: keep continue to support this bad feature, in the hope someone improves it. Or say: fuck it, let's remove it, and if someone comes up with a better idea to do shares, let's add that. This PR goes for the second.

## Description

Remove the option to buy/sell shares. Bankruptcy is still in there, of course.

Mind you: as savegames contain table-headers these days, there is no need for a savegame bump. Simply no longer loading/saving the entries is sufficiently. There is need for a savegame bump between releases, to give users early feedback something is not right, but that is covered already by other PRs.

## Limitations

The money you invested is lost.

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

File: src/company_gui.cpp [hunk 1]
```diff
@@ -2744,25 +2680,6 @@ struct CompanyWindow : Window
 	void OnInvalidateData(int data = 0, bool gui_scope = true) override
```
Comment (2TallTyler) [inline src/company_gui.cpp]: No need for this override to exist anymore?

**Review by 2TallTyler — COMMENTED** [review]
To add to your point that this is a useless feature for players:

By the time a company is available to trade shares and your company is making enough money to afford 25% of that other company, both companies will almost certainly be making a steady profit and well on their way to the "making money faster than you can spend it" runaway profits part of the game. This means that while your investment is almost certain to pay off, you won't need the cash.

The only use I've ever found for shares is bragging rights in network games. Which...isn't really that much fun anyway.

I would like to eventually implement a better share trading system once #8420 is implemented and we have actual clients. That client (player character) could instead be the investor who gets paid dividends or salary from their company to invest in the stock market. Really, the feature I'd want with that would be the ability to sell your company and use the payout to start your next company, like a real entrepreneur. But this feature is a long ways off, so there's no use waiting.

**Review by 2TallTyler — DISMISSED** [review]

**Comment by PeterN** [comment]
I suppose if you load a savegame where shares are in use you're out of luck getting money back.

**Comment by LordAro** [comment]
>Remove: remove company shares from the game

Remove remove

**Review by 2TallTyler — DISMISSED** [review]

**Review by 2TallTyler — APPROVED** [review]

**Comment by LordAro** [comment]
A few issues shares have caused over the years:
https://github.com/OpenTTD/OpenTTD/issues/274
https://github.com/OpenTTD/OpenTTD/issues/459
https://github.com/OpenTTD/OpenTTD/issues/494
https://github.com/OpenTTD/OpenTTD/issues/1590
https://github.com/OpenTTD/OpenTTD/issues/3561
https://github.com/OpenTTD/OpenTTD/issues/3654
https://github.com/OpenTTD/OpenTTD/issues/3663
https://github.com/OpenTTD/OpenTTD/issues/5379
https://github.com/OpenTTD/OpenTTD/issues/5842
https://github.com/OpenTTD/OpenTTD/issues/5947
https://github.com/OpenTTD/OpenTTD/issues/6631
https://github.com/OpenTTD/OpenTTD/issues/7691 <-- the main reason why shares are fundamentally "broken" (which is actually a dup of #494 ?)
https://github.com/OpenTTD/OpenTTD/issues/9281

And some attempts to fix it:
https://github.com/OpenTTD/OpenTTD/issues/1077
https://github.com/OpenTTD/OpenTTD/issues/1445
https://github.com/OpenTTD/OpenTTD/issues/1606
https://github.com/OpenTTD/OpenTTD/issues/1850
https://github.com/OpenTTD/OpenTTD/issues/5857
https://github.com/OpenTTD/OpenTTD/pull/7290
https://github.com/OpenTTD/OpenTTD/pull/7780
https://github.com/OpenTTD/OpenTTD/pull/9300
https://github.com/OpenTTD/OpenTTD/pull/9349
https://github.com/OpenTTD/OpenTTD/pull/9768
https://github.com/OpenTTD/OpenTTD/pull/9770
https://github.com/OpenTTD/OpenTTD/pull/9808
https://github.com/OpenTTD/OpenTTD/pull/9901
https://github.com/OpenTTD/OpenTTD/pull/9905
https://github.com/OpenTTD/OpenTTD/pull/9921
https://github.com/OpenTTD/OpenTTD/pull/10612

**Comment by Drury27** [comment]
I guess I'm the only person who liked this, so I'll give my 2 cents as to why I did:

AAAHogEx is impossible to catch up to and buy without first buying shares in other AIs and then strategically selling them at a later date for profit to be able to bite another piece off the big hog. I had a very cool and fun game where I did that. It reminded me of Offworld Trading Company, which works very much the same. I even thought it might make for an interesting competitive mode if anyone looked into making it possible to do to players (wasn't gonna be me though, hah).

Though if it causes you longterm pain, and since I seem to be the only person missing this, I guess ultimately it's alright that it's gone. Even if I (and only I, seemingly) think it's a shame.

**Pull request merged**